webserver/webserver/src/main/java/io/helidon/webserver/BareResponseImpl.java

```diff
@@ -79,27 +78,28 @@
     private volatile DefaultHttpResponse response;
     private volatile boolean lengthOptimization;
     private volatile boolean isWebSocketUpgrade = false;
+    private CompletableFuture<?> prev;
 
     /**
      * @param ctx the channel handler context
      * @param request the request
      * @param requestContentConsumed whether the request content is consumed
-     * @param thread the outbound event loop thread which will be used to write the response
+     * @param prev Future that represents previous request completion for HTTP pipelining
      * @param requestId the correlation ID that is added to the log statements
      */
     BareResponseImpl(ChannelHandlerContext ctx,
                      HttpRequest request,
                      BooleanSupplier requestContentConsumed,
-                     Thread thread,
+                     CompletableFuture<?> prev,
```

Comment:
Readability nit: `prev` should probably be something like `previousRequestFuture` to "look like", e.g., `responseFuture`, `headersFuture`, etc.

Reply:
Good point